pkg/resources/envoy/deployment.go

```diff
@@ -28,72 +28,90 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func (r *Reconciler) deployment(log logr.Logger) runtime.Object {
+func deploymentName(envoyConfig *v1beta1.EnvoyConfig) string {
+	if envoyConfig.Id == envoyGlobal {
+		return envoyDeploymentName
+	} else {
+		return fmt.Sprintf("%s-%s", envoyDeploymentName, envoyConfig.Id)
+	}
+}
 
-	exposedPorts := getExposedContainerPorts(r.KafkaCluster.Spec.ListenersConfig.ExternalListeners, r.KafkaCluster.Spec.Brokers)
-	volumes := []corev1.Volume{
-		{
-			Name: envoyVolumeAndConfigName,
-			VolumeSource: corev1.VolumeSource{
-				ConfigMap: &corev1.ConfigMapVolumeSource{
-					LocalObjectReference: corev1.LocalObjectReference{Name: envoyVolumeAndConfigName},
-					DefaultMode:          util.Int32Pointer(0644),
+func (r *Reconciler) deployment(log logr.Logger, envoyConfig *v1beta1.EnvoyConfig) runtime.Object {
+	return &appsv1.Deployment{
```

Comment:
can we also add `annotations` to this deployment so it reacts to any change in envoy config?
Similar to `pkg/resources/cruisecontrol/cruisecontrol.go`

Reply:
Yes, I was thinking to implement such an approach. My only concern is that this PR is already addressing more than 1 problem (multiple envoys & un-managed LB).
@amuraru, what do you think about using another PR for this?

Reply:
+1 on addressing this in a new PR